lib/lists.nix

```
@@ -637,6 +637,76 @@ rec {
        else if start + count > len then len - start
        else count);
 
+  /* The length of the common prefix of two lists.
+
+  Type: commonPrefixLength :: [a] -> [a] -> Int
+
+  Example:
+    commonPrefixLength [ 1 2 3 4 5 6 ] [ 1 2 4 8 ]
+    => 2
+    commonPrefixLength [ 1 2 3 ] [ 1 2 3 4 5 ]
+    => 3
+    commonPrefixLength [ 1 2 3 ] [ 4 5 6 ]
+    => 0
+  */
+  commonPrefixLength =
+    let
+      shorterFirst =
+        shorter: longer:
+        let
+          # We fold over the shorter list, no need to check elements of the
+          # longer list that couldn't be matched by the shorter one
+
+          # index invariant:
+          # - if index < 0 then (- index - 1) is the first index that differs between the two lists
+          # - if index >= 0 then all elements below that index match and s == elemAt shorter index
```

```suggestion
          # - if index >= 0 then all elements below that index match and `s == elemAt shorter index` holds
```

That was hard to parse otherwise